builtin/playerinfo.lua

```diff
@@ -0,0 +1,139 @@
+-- Minetest: builtin/playerinfo.lua
+
+playerinfo = {}
+
+--
+-- Player data handler
+--
+
+local playerdata = {}
+
+function playerinfo.set_data(player, data)
+	local pname = player
+	if type(pname) ~= "string" then
+		pname = player:get_player_name()
+	end
+	-- Set defaults if player data doesn't exist
+	if not playerdata[pname] then
+		playerdata[pname] = {icon = "logo.png", description = "No information available."}
+	end
+	-- Set icon
+	if data and data.icon then
+		playerdata[pname].icon = data.icon
+	end
+	-- Set description
+	if data and data.description then
+		playerdata[pname].description = data.description
+	end
+end
+
+function playerinfo.get_data(player)
+	local pname = player
+	if type(pname) ~= "string" then
+		pname = player:get_player_name()
+	end
+	-- Set defaults if player data doesn't exist
+	if not playerdata[pname] then
+		playerdata[pname] = {icon = "logo.png", description = "No information available."}
+	end
+
+	return playerdata[pname]
+end
+
+--
+-- Player list handler
+--
+
+-- Stores each player's player list, to avoid changes while the formspec is refreshed
+local formspec_players = {}
+
+local function formspec_players_set(name)
+	formspec_players[name] = {}
+	for i, player in pairs(minetest.get_connected_players()) do
+		-- Add player name
+		local pname = player:get_player_name()
+		-- Add player ping
+		local pping = minetest.get_player_information(pname).avg_rtt / 2
+		pping = math.floor(pping * 1000) -- Milliseconds
+
+		table.insert(formspec_players[name], {name = pname, ping = pping})
+	end
+end
+
+local function formspec_string(name, index)
+	local formspec="size[8,6]"
+
+	-- Turn the player table into a string
+	local players = ""
+	for i, entry in pairs(formspec_players[name]) do
+		local ping_color = "#AAFFAA"
+		if entry.ping > 500 then
+			ping_color = "#FFAAAA"
+		elseif entry.ping > 100 then
+			ping_color = "#FFFF0AA"
+		end
+
+		players = players..ping_color..entry.name.." ("..entry.ping.."ms)"
+		if i < #formspec_players[name] then
+			players = players..","
+		end
+	end
+	formspec = formspec.."label[4,0;Players: "..#formspec_players[name].." / "..minetest.setting_get("max_users").."]"
```

Comment:
Nope, builtin is run on the server. The only Lua environment on the client is for the maim menu.